src/helpers/proxy.ts

```diff
@@ -0,0 +1,6 @@
+import { ProxyAgent } from 'undici';
+import { UploaderArgs, UploaderEnvs } from '../types';
+
+export function addProxyIfNeeded(envs: UploaderEnvs, args: UploaderArgs): ProxyAgent | undefined {
```

Comment:
I'm not sure this needs to be in it's own file. It's only being used in `web`

Reply:
I thought it was easier to locate it if it needed to be changed, but I can put it back in web if you would like.

Reply:
we could put the `generateHeaders` function here then if we want proxy stuff to be in one file.